map/framework.cpp

```diff
@@ -2489,15 +2489,14 @@ void Framework::SetRouterImpl(RouterType type)
       return m_model.GetIndex().GetMwmIdByCountryFile(CountryFile(countryFile)).IsAlive();
     };
 
-    //    @TODO(bykoianko, dobriy-eeh). Pass numMwmIds to router and CrossMwmIndexGraph.
-    //    shared_ptr<routing::NumMwmIds> numMwmIds = make_shared<routing::NumMwmIds>();
-    //    m_storage.ForEachCountryFile([&](platform::CountryFile const & file){
-    //      numMwmIds->RegisterFile(file);
-    //    });
+    auto numMwmIds = make_shared<routing::NumMwmIds>();
+    m_storage.ForEachCountryFile([&](platform::CountryFile const & file){
```

Comment:
In the direct access case variable is on the stack, therefore there's no cache miss at all. In the indirect access, there may be a cache miss (not more than one in this case), as you accessing memory from the pool.